src/parser/Parser.ts

```diff
@@ -2718,6 +2727,70 @@ export class Parser {
         return this.tokens[idx - 1];
     }
 
+    /**
+     * Finds the previous identifier in a chain (e.g. 'm.obj.func(someFunc()).value'), skipping over any arguments of function calls
+     * If this function was called with the token at 'value' above, the previous identifier in the chain is 'func'
+     * @param currentToken token to start from
+     * @param allowCurrent can the current token be the token that's the identifier?
+     * @returns the previous identifer
+     */
+    public getPreviousIdentifierInChain(currentToken: Token, allowCurrent = false): Token {
+        if (!allowCurrent) {
+            currentToken = this.getPreviousToken(currentToken);
```

Comment:
`getPreviousToken` does a `parser.tokens.indexOf(token)` so this function may be very inefficient in very long scripts. You should maybe get the index of the token, then walk back from it?